src/twisted/mail/_except.py

```diff
@@ -138,17 +138,20 @@ def __init__(self, code, resp, log=None, addresses=None, isFatal=False,
 
     if _PY3:
         def __str__(self):
-            return nativeString(bytes(self))
+            return self.__bytes__().decode("utf-8")
     else:
         def __str__(self):
             return self.__bytes__()
 
 
     def __bytes__(self):
         if self.code > 0:
-            res = [networkString("%.3d %s" % (self.code, self.resp))]
+            res = [u"{:03d} {}".format(self.code, self.resp).encode("utf-8")]
```

Comment:
I don't see how this can be done easily.  These exceptions don't have access to self._encoding, so every single exception inheriting from SMTPClientError needs to be modified to take an encoding argument in its constructor....and there are lots.

I'm relying on this code in SMTPClient.sendLine() to fail if any invalid characters are in the stream:
```
        try:
            line.decode(self._encoding)
        except UnicodeDecodeError:
            self.sendError(
                SMTPProtocolError(-1,
                    "Server supports {} encoding, invalid "
                    "characters detected in sent line:\n{} .".format(
                        self._encoding, repr(line)),
                    self.log.str()))
            return
```

I've used that successfully to test sending invalid ASCII characters to a Sendmail server which does not support SMTPUTF8.